### PR TITLE
RNA: Connection Screen

### PR DESCRIPTION
**Pull request by sergeymitr**
#### Changes proposed in this Pull Request:
- add `ConnectScreen` component implements the connection screen
- add `ImageSlider` component for the image sliding/rotation on the right side of the connection screen
- integration of the new `ConnectScreen` component into the Connection Manager UI
- moving the `getRedirectUrl()` function into the RNA Components package

#### Jetpack product discussion
1200295095943283-as-1200465631534665
p1HpG7-caw-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_If you're using a JN site, you need to install [this plugin](https://gist.github.com/sergeymitr/bc005e1b893e778c03e4f43debb91c30) to activate the Connection Manager page._

1. Disconnect Jetpack.
2. Go to "Tools -> Connection Manager".
3. Confirm that the page looks like this:
Screen Shot 2021-07-13 at 4 38 16 PM
4. Click "Set up Jetpack". The button should become inactive, then you should be redirected to Calypso connection flow.
5. Approve the connection in Calypso. Confirm that you're redirected back to the Connection Manager, and the page says "Site and User Connected".

### CODE REVIEW

**Comment by sdixon194** [comment]
@sergeymitr Since code freeze is upon us, are there plans to make this into 9.9, or can we defer this to the next release and remove it from this milestone?

**Comment by sergeymitr** [comment]
Hi @sdixon194, thanks for the ping!
Moved it to 10.0.

**Review by fgiannar — COMMENTED** [review]
Hey Sergey, great work here! Well done!

I tested it using the provided instructions and it works as expected.
I also made sure to check the UI in Chrome (latest), Firefox (latest) and Edge.

Some UI related comments:

- Checkmark alignment seems a bit off:
Screenshot 2021-07-07 at 12 34 05 PM

- Is this the intended color on hover?:
Screenshot 2021-07-07 at 12 35 36 PM

- The footer section: "By signing up you agree to terms etc etc" seems to be missing compared to the Design P2 : p1HpG7-caw-p2
Is this intended?

- The font seems also a bit different than the one provided in the Design P2.
-  The image seems to be overflowing its container:
Screenshot 2021-07-07 at 1 22 26 PM


Finally, a question: The usage of the `statusCallback` property is not 100% clear to me. Why do we need this?

I'd suggest also getting a Review from the Design team as well if you agree :)

**Comment by sergeymitr** [comment]
Hi @fgiannar, thanks for the review!

> The usage of the statusCallback property is not 100% clear to me. Why do we need this?

This is a way for a package consumer to pull the connection status from inside of the package.
For example, `connection-ui` doesn't pull connection status itself, and relies on the connection package for that data.
Then it assigns the status into its own state, so it can be reused outside of the package.
This is a trick that can save developers quite a bit of headache:
https://github.com/Automattic/jetpack/blob/update/rna-connect-screen/projects/packages/connection-ui/_inc/components/admin/index.jsx#L31-L36

File: projects/js-packages/connection/components/connect-screen/index.jsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+					from={ from }
+					redirectUri={ redirectUri }
+					statusCallback={ statusCallback }
+					connectLabel={ __( 'Set up Jetpack', 'jetpack' ) }
+				/>
+
+				<div className="jp-connnect-screen--tos">
+					{ createInterpolateElement(
+						__(
+							'By clicking the button below, you agree to our <tosLink>Terms of Service</tosLink> and to <shareDetailsLink>share details</shareDetailsLink> with WordPress.com.',
```
Comment (fgiannar) [inline projects/js-packages/connection/components/connect-screen/index.jsx:75]:
```suggestion
							'By clicking the button above, you agree to our <tosLink>Terms of Service</tosLink> and to <shareDetailsLink>share details</shareDetailsLink> with WordPress.com.',
```

**Review by fgiannar — DISMISSED** [review]
Hi Sergey!

Great work here! I've tested again and all looks great!

Added a very minor suggestion that we can address on a follow-up PR though.

Let's 🚢 !

File: projects/js-packages/connection/components/connect-screen/index.jsx (hunk 1 above)
Comment (sergeymitr) [reply projects/js-packages/connection/components/connect-screen/index.jsx]: Good catch, thanks!

**Review by fgiannar — APPROVED** [review]
Many thanks for addressing my suggestion so quickly!
Re-approving :shipit:

**Pull request merged**